### PR TITLE
Martin/chore refactor

### PR DESCRIPTION
**Pull request by martinohansen**
This is the refactoring from https://github.com/martinohansen/ynabber/pull/36

### CODE REVIEW

**Comment by hpernu** [comment]
Looks fine and nothing in functionality is supposedly changed.

However, a test is failing so please either fix whatever is causing it or fix tests. Adding tests is not a bad idea anyway if we want to be certain this does not change anything. ( Automated tests are especially useful for refactoring for this reason. )

Because of failing tests, I would not approve this at this point.
Once tests pass too, I can approve this, although you can also self approve then.

**Comment by martinohansen** [comment]
> Looks fine and nothing in functionality is supposedly changed.
> 
> However, a test is failing so please either fix whatever is causing it or fix tests. Adding tests is not a bad idea anyway if we want to be certain this does not change anything. ( Automated tests are especially useful for refactoring for this reason. )
> 
> Because of failing tests, I would not approve this at this point. Once tests pass too, I can approve this, although you can also self approve then.

That should be fixed now, please look again.

File: config.go [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
 
+const DateFormat = "2006-01-02"
+
+type Date time.Time
+
+// Decode implements `envconfig.Decoder` for Date to parse string to time.Time
+func (date *Date) Decode(value string) error {
+	time, err := time.Parse(DateFormat, value)
+	if err != nil {
+		return err
```
Comment (hpernu) [inline config.go:16]: Does the user actually see a sensible error message?
I.e. unable to convert this and this to date?

File: config.go [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+	return nil
+}
+
+type AccountMap map[string]string
+
+// Decode implements `envconfig.Decoder` for AccountMap to decode JSON properly
+func (accountMap *AccountMap) Decode(value string) error {
+	err := json.Unmarshal([]byte(value), &accountMap)
+	if err != nil {
+		return err
```
Comment (hpernu) [inline config.go:28]: Same here about the sensible error message.

**Review by hpernu — APPROVED** [review]
Put a couple of comments but they are not blockers. Merely interested whether the error messages will be informative for the user when decode fails.

File: config.go (hunk 2 above)
Comment (martinohansen) [reply config.go:28]: It fails somewhat nice
```
$ go run cmd/ynabber/main.go
2022/11/17 17:17:34 envconfig.Process: assigning NORDIGEN_NORDIGEN_ACCOUNTMAP to AccountMap: converting 'fooo' to type ynabber.AccountMap. details: invalid character 'o' in literal false (expecting 'a')
exit status 1
```

**Pull request merged**